be/src/vec/sink/vmemory_scratch_sink.cpp

```diff
@@ -75,13 +76,18 @@ Status MemoryScratchSink::prepare(RuntimeState* state) {
     return Status::OK();
 }
 
-Status MemoryScratchSink::send(RuntimeState* state, Block* block, bool eos) {
-    if (nullptr == block || 0 == block->rows()) {
+Status MemoryScratchSink::send(RuntimeState* state, Block* input_block, bool eos) {
+    if (nullptr == input_block || 0 == input_block->rows()) {
         return Status::OK();
     }
     std::shared_ptr<arrow::RecordBatch> result;
+    // Exec vectorized expr here to speed up, block.rows() == 0 means expr exec
+    // failed, just return the error status
+    Block block;
+    RETURN_IF_ERROR(VExprContext::get_output_block_after_execute_exprs(_output_vexpr_ctxs,
```

Comment:
This line is the core of this PR, right?

Reply:
yes